src/lib.rs

```diff
@@ -763,4 +776,29 @@ mod tests {
         non_empty.head *= 42;
         assert_eq!(non_empty.head, 42);
     }
+
+    #[cfg(feature = "serialize")]
+    #[test]
+    fn test_simple_serializable() {
+        // Given
+        let non_empty = NonEmpty::new(SimpleSerializable(42));
+
+        // When
+        let res = serde_json::to_string(&non_empty);
+
+        // Then
+        assert_eq!(res.ok(), Some("{\"head\":42,\"tail\":[]}".into()));
+    }
+
+    #[cfg(feature = "serialize")]
+    #[test]
+    fn test_simple_deserializable() {
+        // Given
+        let json = "{\"head\":42,\"tail\":[]}";
+
+        // When
+        let res = serde_json::from_str::<'_, NonEmpty<SimpleSerializable>>(json);
+        // Then
+        assert_eq!(res.ok(), Some(NonEmpty::new(SimpleSerializable(42))));
+    }
```

Comment:
I think having a single test that does a roundtrip would be better than two separate tests. Something like:
```rust
assert_eq!(original_val, deserialize(serialize(original_val)));
```

Reply:
I was actually thinking of a round-trip as a third test.  But it's probably not worth the time to debate.  I'll just combine them to the 1 as you suggest.